paasta_tools/kubernetes/wrappers.py

```diff
@@ -0,0 +1,121 @@
+import logging
+from abc import ABC
+from abc import abstractmethod
+
+from kubernetes.client import V1DeleteOptions
+from kubernetes.client.rest import ApiException
+
+from paasta_tools.kubernetes_tools import get_deployment_config
+from paasta_tools.kubernetes_tools import KubeClient
+from paasta_tools.kubernetes_tools import KubeDeployment
+from paasta_tools.kubernetes_tools import KubernetesDeploymentConfig
+from paasta_tools.utils import SystemPaastaConfig
+
+
+class Application(ABC):
+    def __init__(
+            self,
+            item: KubeDeployment,
+            kube_client: KubeClient,
```

Comment:
IMO if there are no strong reasons to have `kube_client ` as a member, it's better not to do that and just pass it as an arg into `deep_delete()`/`delete_pod_disruption_budget ()`.

Reply:
I just don't like to write kube_client as an argument whenever a function call is made. What about adding a function set_kube_client, and make it optional in __init__ instead?

Reply:
@EmanekaT the problem here is that `Application` doesn't "own" `kube_client`, but just use a provided `kube_client` to connect to Kubernetes, in that way `set_kube_client` isn't better.
Just imaging that we're using a pool of `kube_client` while deleting applications, in such case the process looks very natural: we're getting the next available `kube_client` from the pool and call `deep_delete()` with it, this way the whole process will be elegantly parallelised.  And now imagine how that will look for saved `kube_client`s.

Reply:
Gotcha. I didn't consider multi-threading.